gratipay/models/participant/email.py

```diff
@@ -106,6 +103,16 @@ def validate_email_verification_request(self, c, email, *packages):
         if not all(email in p.emails for p in packages):
             raise EmailNotOnFile()
 
+        unverified_email = c.one("""
+            SELECT address
```

Comment:
A user could repeatedly add an email + remove it to circumvent this restriction, no? The throttling strategy we use will have to rely on previous state (as you've mentioned in the ticket description)